hack/verify-golangci-lint.sh

```diff
@@ -18,18 +18,27 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-VERSION=v1.42.0
+VERSION=v1.43.0
 URL_BASE=https://raw.githubusercontent.com/golangci/golangci-lint
 URL=$URL_BASE/$VERSION/install.sh
+INSTALL_CHECKSUM=294771225087ee48c8e0a45a99ac82ed8f9c6e9d384e692ab201986479c8594f
```

Comment:
Hey @vinayakankugoyal thanks for adding this. Check. Expanding a little on @cpanato's comment, could you add a comment so that anyone bumping the golangci-lint version can know how to obtain the install script sha?

Reply:
Done.